### PR TITLE
Set the number of workers to 0 for validation and test sets in all enc-dec models

### PR DESCRIPTION
**Pull request by MaximumEntropy**
Signed-off-by: MaximumEntropy <sandeep.subramanian.1@umontreal.ca>

# What does this PR do ?

Sets the number of data workers to 0 for validation and test sets for all encoder-decoder models to sidestep a strange CUDA error.

NOTE: This is not a permanent solution and just a temporary fix.

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py
Comment (SeanNaren) [inline nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py:830]: Is it worth throwing a warning or do you think this will be a relatively temporary fix?

**Review by SeanNaren — APPROVED** [review]
LGTM, curious what the error is!

**Comment by titu1994** [comment]
Oh that error is back, you can easily try it by seting asr model transcribe call with num_workers > 0. It works on some Ubuntu machines just fine, but throws pretty much the same error on others and on Mac's

**Comment by MaximumEntropy** [comment]
@SeanNaren and @titu1994, we chatted with Piotr from the frameworks team and he mentioned that this is likely something else in our containers/env that is causing this break - possibly cuDF. He said he'll take a look and see if there is a cleaner fix so until then I'm blocking merging.

**Review by ericharper — CHANGES_REQUESTED** [review]
Blocking while we try to root cause this further.

**Review by ericharper — APPROVED** [review]
Approving this work-around until we find a better solution.

**Pull request merged**
